packages/editor/src/lib/editor/Editor.ts

```diff
@@ -7349,6 +7349,49 @@ export class Editor extends EventEmitter<TLEventMap> {
 		return this
 	}
 
+	/**
+	 * Fit a frame to its content.
+	 *
+	 * @param id - Id of the frame you wish to fit to content.
+	 *
+	 * @public
+	 */
+	fitFrameToContent(id: TLShapeId): this {
+		const frame = this.getShape<TLFrameShape>(id)
+		if (!frame) return this
+
+		const childIds = this.getSortedChildIdsForParent(frame.id)
+		const bounds = Box2d.Common(childIds.map((s) => this.getShapePageBounds(s)!))
```

Comment:
i think this will fail for nested frames or frames inside groups (i.e. frames whose parents are not the page) and rotated frames. Probably need to construct the children's bounds in the frame's space and then use that as a basis for calculating the size adjustments and offsets